Add issues closed pie chart and DatePickerRange

christinaexyou opened the pull request:
@cdolfi Resolves the rest of #468 and #469 by making the following changes:

* Adds pie for issues closed
* Adds DatePickerRange to allow user to pick start and end date for the data which pie chart captures
* Renames variables in `process_data` for better readability

christinaexyou commented:
@cdolfi

cdolfi commented on pages/chaoss/chaoss.py:
```diff
@@ -4,23 +4,52 @@
 # import visualization cards
+from .visualizations.contrib_drive_repeat import gc_contrib_drive_repeat
+from .visualizations.first_time_contributions import gc_first_time_contributions
+from .visualizations.contributors_types_over_time import gc_contributors_over_time
```
No changes should be on the file

cdolfi commented on pages/chaoss/visualizations/contrib_importance_pie.py:
```diff
@@ -70,6 +71,7 @@
                                                 {"label": "Commit", "value": "Commit"},
                                                 {"label": "Issue Opened", "value": "Issue Opened"},
+                                                {"label": "Issue Closed", "value": "Issue Closed"},
                                                 {"label": "PR Open", "value": "PR Open"},
```
Add option for "Issue Comment" or group together pr and issue comment as "Comment". example of this can be seen in contributors/contrib_by_action.py. Line of code that makes it work: 

`df = df[df["Action"].str.contains(action)]`

cdolfi commented on pages/chaoss/visualizations/contrib_importance_pie.py:
```diff
@@ -212,51 +233,66 @@ def create_top_k_cntrbs_graph(repolist, action_type, patterns, top_k):
+    df = df.sort_values(by='created_at', ascending=True)
+
+    # if the start_date and/or the end date is not specified set them to the beginning and most recent created_at date
+    if start_date is None:
```
Can do this more concise: 

`# filter values based on date picker
    if start_date is not None:
        df = df[df.created_at >= start_date]
    if end_date is not None:
        df = df[df.created_at <= end_date]`

cdolfi commented on pages/chaoss/visualizations/contrib_importance_pie.py:
```diff
@@ -212,51 +233,66 @@ def create_top_k_cntrbs_graph(repolist, action_type, patterns, top_k):
+
     # subset the df such that it only contains rows where the Action column value is the action type
-    df = df[df["Action"].str.contains(action_type)]
+    df_final = df_in_range[df_in_range["Action"].str.contains(action_type)]
```
Why change to df_final rather than reassigning df?

christinaexyou replied:
For better readability but I'll change it back to df

cdolfi reviewed (changes requested):
Implement changes and run to make sure it builds. I am getting errors if I try to run the app

christinaexyou commented:
@cdolfi Changes made:
* Syntax to subset df by start and end date
* Added `Issue Comment` to dropdown menu
* Reverted `df_final` to `df` 
* Dropped `chaoss.py` file

cdolfi commented:
@christinaexyou Once you rebase this will be ready to merge!
